### PR TITLE
Issue #12572: fix bump_license_year.yml not adding year to commit message

### LINKED ISSUE #12572: `bump_license_year.yml` not adding year to commit message.
Noted at https://github.com/checkstyle/checkstyle/commit/54616f662130aa0ca5afe45f72d36cdefb78a781, the commit message didn't include the year, however, the changes made by the commit are correct.

The bad commit message was:
> minor: Bump year to 

It should have been:
> minor: Bump year to 2023

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #12572 

Source: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Proof of working: https://github.com/Vyom-Yadav/actions-test/commit/8ae3f4490e16f3bbba5f252f96b03e15198e8122

### CODE REVIEW

File: .github/workflows/bump_license_year.yml [hunk 1]
```diff
@@ -20,14 +20,11 @@ jobs:
       - name: Checkout the latest code
         uses: actions/checkout@v3
       - name: Set Current Year
-        id: CURRENT_YEAR
         run: |
-          echo "year=$(date +'%Y')" >> "$GITHUB_OUTPUT"
+          echo "YEAR=$(date +'%Y')" >> "$GITHUB_ENV"
```
Comment (rnveach) [inline .github/workflows/bump_license_year.yml:24]: Where is `GITHUB_ENV` going anyways? I don't see a usage here. Is this somehow connected to `env`?

**Review by rnveach — APPROVED** [review]
Looks good to me, but I don't really know this too well. However....

File: .github/workflows/bump_license_year.yml (hunk 1 above)
Comment (Vyom-Yadav) [reply .github/workflows/bump_license_year.yml:24]: Yes it is, see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

**Review by romani — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
@nrmancuso , ups , did not notice you as reviewer

**Pull request merged**